src/mapgen_flat.cpp

```diff
@@ -453,6 +453,8 @@ MgStoneType MapgenFlat::generateBiomes(float *heat_map, float *humidity_map)
 					stone_type = DESERT_STONE;
 				else if (biome->c_stone == c_sandstone)
 					stone_type = SANDSTONE;
+			} else {
+				continue;
```

Comment:
Yes, but not all of the conditionals make use of `biome`.
Additionally, that continue statement will skip incrementing the voxelmanip index, resulting in mapgen breakage.